src/containers/studies/StudiesSagas.js

```diff
@@ -269,17 +266,32 @@ function* deleteStudyParticipantWorker(action :SequenceAction) :Generator<*, *,
 
     const { studyId, participantEntityKeyId, participantId } = action.value;
 
-    const { response, timeout } = yield race({
-      response: call(ChronicleApi.deleteStudyParticipant, participantId, studyId),
-      timeout: delay(1000 * 10) // 10 seconds
-    });
-    if (response && response.error) throw response.error;
+    yield call(ChronicleApi.deleteStudyParticipant, participantId, studyId);
 
     yield put(deleteStudyParticipant.success(action.id, {
       participantEntityKeyId,
       studyId,
-      timeout: isDefined(timeout)
     }));
+
+    // update association with status = 'DELETE'
+    const participatedInESID = yield select(selectEntitySetId(PARTICIPATED_IN));
+    const statusPTID = yield select(selectPropertyTypeId(STATUS));
+
+    const participatedInEKID = yield select(
+      (state) => state.getIn(['studies', 'participants', studyId, participantEntityKeyId, PARTICIPATED_IN_EKID, 0])
+    );
+
+    const response = yield call(updateEntityDataWorker, updateEntityData({
+      entities: {
+        [participatedInEKID]: {
+          [statusPTID]: [DELETE],
+        }
+      },
+      entitySetId: participatedInESID,
+      updateType: UpdateTypes.PartialReplace
+    }));
+    if (response.error) throw response.error;
```

Comment:
This is an edge case ... if the above delete succeeds but this update fails, you won't get another chance to make this update. Will things break if this happens?

Reply:
Great question. An alternative approach might be to reverse the calls (make the update, then call delete), but that approach also has a similar loop hole: if the update succeeds and the delete fails, then we will have erroneously marked the study as deleted, and the user won't get another chance to actually delete

The current approach of updating after a successful delete job scheduling is arguably better, but not perfect. If the update fails, the user might still view the study but it will eventually disappear from the UI once the delete job is finished.